### PR TITLE
rgw: format logs in file rgw_lc.cc

### PR DESCRIPTION
**Pull request by qrGitHub**

### CODE REVIEW

**Comment by cbodley** [comment]
jenkins test docs

**Comment by qrGitHub** [comment]
Hi @cbodley, can this be merged?

**Comment by qrGitHub** [comment]
Hi @joscollin, would you help to merge this? Thanks.

**Comment by joscollin** [comment]
jenkins test docs

File: src/rgw/rgw_lc.cc [hunk 1]
```diff
@@ -351,13 +351,13 @@ int RGWLC::bucket_lc_process(string& shard_id)
   string bucket_id = result[2];
   int ret = store->get_bucket_info(obj_ctx, bucket_tenant, bucket_name, bucket_info, NULL, &bucket_attrs);
   if (ret < 0) {
-    ldout(cct, 0) << "LC:get_bucket_info failed" << bucket_name <<dendl;
+    ldout(cct, 0) << "LC:get_bucket_info for " << bucket_name << " failed" << dendl;
     return ret;
   }
 
   ret = bucket_info.bucket.bucket_id.compare(bucket_id) ;
   if (ret !=0) {
```
Comment (joscollin) [inline src/rgw/rgw_lc.cc]: put space after the `!=`

File: src/rgw/rgw_lc.cc [hunk 2]
```diff
@@ -351,13 +351,13 @@ int RGWLC::bucket_lc_process(string& shard_id)
...
-    ldout(cct, 0) << "LC:old bucket id find, should be delete" << bucket_name <<dendl;
+    ldout(cct, 0) << "LC:old bucket id find, " << bucket_name << " should be deleted" << dendl;
```
Comment (joscollin) [inline src/rgw/rgw_lc.cc]: I think this is better: 
`LC:old bucket id found.  " << bucket_name << " should be deleted.`

File: src/rgw/rgw_lc.cc [hunk 3]
```diff
@@ -709,7 +709,8 @@ int RGWLC::process(int index, int max_lock_secs)
     entry.second = lc_processing;
     ret = cls_rgw_lc_set_entry(store->lc_pool_ctx, obj_names[index],  entry);
     if (ret < 0) {
-      dout(0) << "RGWLC::process() failed to set obj entry " << obj_names[index] << entry.first << entry.second << dendl;
+      dout(0) << "RGWLC::process() failed to set obj entry " << obj_names[index] <<
+          "(" << entry.first << "," << entry.second << ")" << dendl;
```
Comment (joscollin) [inline src/rgw/rgw_lc.cc]: Put space before `(`

**Comment by qrGitHub** [comment]
@joscollin, changes are completed.

File: src/rgw/rgw_lc.cc [hunk 4]
```diff
@@ -351,13 +351,13 @@ int RGWLC::bucket_lc_process(string& shard_id)
...
-    ldout(cct, 0) << "LC:get_bucket_info failed" << bucket_name <<dendl;
+    ldout(cct, 0) << "LC:get_bucket_info for " << bucket_name << " failed" << dendl;
     return ret;
   }
 
   ret = bucket_info.bucket.bucket_id.compare(bucket_id) ;
-  if (ret !=0) {
-    ldout(cct, 0) << "LC:old bucket id find, should be delete" << bucket_name <<dendl;
+  if (ret != 0) {
+    ldout(cct, 0) << "LC:old bucket id found. " << bucket_name << " should be deleted" << dendl;
```
Comment (joscollin) [inline src/rgw/rgw_lc.cc]: This check should be `if (ret == 0)` in-order to find an old bucket id, right ?
@cbodley Could you please re-check this ?

Comment (qrGitHub) [reply src/rgw/rgw_lc.cc]: No, the check is ok.

The compare here is to verify the 'bucket_id' in debug.rgw.log/lc.<index> is valid.
Failed compare means the 'bucket_id' in debug.rgw.log/lc.<index> is invalid, when we shouldn't run lifecycle for this bucket.
One possible circumstance is the bucket whose id is 'bucket_id' was removed, after which a new bucket with the same name is created.

Perhaps this line is better:
ldout(cct, 0) << "LC:old bucket id found. " << bucket_name << " has been deleted and re-created" << dendl;

Comment (joscollin) [reply src/rgw/rgw_lc.cc]: @qrGitHub ok, thanks for clarifying.

File: src/rgw/rgw_lc.cc [hunk 5]
```diff
@@ -683,7 +683,7 @@ int RGWLC::process(int index, int max_lock_secs)
     cls_rgw_lc_obj_head head;
     ret = cls_rgw_lc_get_head(store->lc_pool_ctx, obj_names[index], head);
     if (ret < 0) {
-      dout(0) << "RGWLC::process() failed to get obj head " << obj_names[index] << ret << dendl;
+      dout(0) << "RGWLC::process() failed to get obj head " << obj_names[index] << ", ret=" << ret << dendl;
```
Comment (theanalyst) [inline src/rgw/rgw_lc.cc]: since we're touching the logs anyway, let's convert all dout calls to ldout, we already have a ctx

Comment (qrGitHub) [reply src/rgw/rgw_lc.cc]: @theanalyst Ok, I will take this later. Would you tell me the difference between dout and ldout? Thanks.

**Comment by qrGitHub** [comment]
change dout to ldout

**Comment by qrGitHub** [comment]
Hi @cbodley @adamemerson, can this be merged? Thanks.

**Comment by yuriw** [comment]
[wip-yuri3-testing-2018-01-09-1802](https://trello.com/c/VlBAyUMv/225-wip-yuri3-testing-2018-01-09-1802)

**Pull request merged**